feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/domain/validations/NotZeroBalanceValidation.kt

```diff
@@ -0,0 +1,20 @@
+package jp.co.soramitsu.feature_staking_impl.domain.validations
+
+import jp.co.soramitsu.common.validation.DefaultFailureLevel
+import jp.co.soramitsu.common.validation.Validation
+import jp.co.soramitsu.common.validation.ValidationStatus
+import java.math.BigDecimal
+
+class NotZeroBalanceValidation<P, E>(
+    val amountExtractor: (P) -> BigDecimal,
+    val errorProvider: () -> E
+) : Validation<P, E> {
+
+    override suspend fun validate(value: P): ValidationStatus<E> {
+        return if (amountExtractor(value) > BigDecimal.ZERO) {
+            ValidationStatus.Valid()
+        } else {
+            ValidationStatus.NotValid(DefaultFailureLevel.WARNING, errorProvider())
+        }
+    }
```

Comment:
It looks like there is no need to make it generic since it is used only once for now. Also, taking into account logic described in comment below, it will be rather specific validation with not really extendable meaning